src/components/screens/vacation-requests-screen.tsx

```diff
@@ -200,6 +199,70 @@ const VacationRequestsScreen = () => {
     setLoading(false);
   };
 
+  /**
+   * Update a vacation request status
+   *
+   * @param status VacationRequestStatus value for the new status
+   * @param selectedRowIds GridRowId[] list of Ids of selected rows(vacation requests)
+   */
+  const updateVacationRequestStatus = async (
+    status: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => {
+    if (!loggedInUser) return;
+
+    try {
+      setLoading(true);
+      const updatedVacationRequests = await Promise.all(
+        selectedRowIds.map(async (vacationRequestId) => {
+          const vacationRequest = vacationRequests.find(
+            (vacationRequest) => vacationRequest.id === vacationRequestId
+          );
+          if (!vacationRequest) return null;
+
+          const statusExists = vacationRequest.status?.some(
+            (existingStatus) => existingStatus.createdBy === loggedInUser.id
+          );
+
+          const newOrUpdatedStatus = {
+            status,
+            createdBy: loggedInUser.id,
+            updatedAt: new Date(),
+            message: `This vacation request is ${status}`
+          };
+
+          const updatedStatus = statusExists
+            ? vacationRequest.status?.map((existingStatus) =>
+                existingStatus.createdBy === loggedInUser.id
+                  ? { ...existingStatus, ...newOrUpdatedStatus }
+                  : existingStatus
+              )
+            : [...(vacationRequest.status || []), newOrUpdatedStatus];
+
+          return await vacationRequestsApi.updateVacationRequest({
```

Comment:
If we already have Promise.all above, maybe we dont need await here as Promise.all should wait for all promises to resolve it
```suggestion
          return vacationRequestsApi.updateVacationRequest({
```